waves_gateway/controller/flask_rest_controller.py

```diff
@@ -114,41 +163,73 @@ def _get_attempt_list(self) -> str:
 
         if trigger_receiver is not None:
             trigger_receiver_as_int = int(trigger_receiver)
 
-        if (trigger_tx is None) and (anything is None):
+        if (trigger_tx is None) and (anything is None) and (attempt is None) and (tries is None):
             return flask_module.abort(400)
 
         query = AttemptListQuery(
             trigger_currency=trigger_currency,
             trigger_tx=trigger_tx,
             trigger_receiver=trigger_receiver_as_int,
-            anything=anything)
+            anything=anything,
+            attempt=attempt,
+            tries=tries)
 
         attempt_lists = self._gateway_controller.query_attempt_lists(query)
 
         serialized_lists = map_array(lambda x: self._serializer.attempt_list_as_dict(x), attempt_lists)
 
         return flask_module.jsonify(serialized_lists)
 
-    def _check_coin_transaction(self, tx: str) -> str:
-        try:
-            self._gateway_controller.check_coin_transaction(tx)
-        except InvalidTransactionIdentifier:
-            flask_module.abort(400)
-        except NotImplementedError:
-            flask_module.abort(501)
-
-        return ''
-
-    def _check_waves_transaction(self, tx: str) -> str:
-        try:
-            self._gateway_controller.check_waves_transaction(tx)
-        except InvalidTransactionIdentifier:
-            flask_module.abort(400)
-
-        return ''
+    def _check_coin_transaction(self, tx: str):
+        key = flask_module.request.headers['api-key']
+        if key is not None and key == self._key:
+            try:
+                self._gateway_controller.check_coin_transaction(tx)
+            except InvalidTransactionIdentifier:
+                flask_module.abort(400)
+            except NotImplementedError:
+                flask_module.abort(501)
+
+            return ''
+        else:
+            flask_module.abort(403)
+
+    def _get_block_heights(self):
+        key = flask_module.request.headers['api-key']
+        if key is not None and key == self._key:
+            heights = self._gateway_controller.get_block_heights()
+            return flask_module.jsonify(heights)
+        else:
+            flask_module.abort(403)
+
+    def _check_waves_transaction(self, tx: str):
+        key = flask_module.request.headers['api-key']
+        if key is not None and key == self._key:
+            try:
+                self._gateway_controller.check_waves_transaction(tx)
+            except InvalidTransactionIdentifier:
+                flask_module.abort(400)
+
+            return ''
+        else:
+            flask_module.abort(403)
+
+    def _trigger_attemptlist_retry(self):
+        key = flask_module.request.headers['api-key']
+        if key is not None and key == self._key:
+            attempt_list_id = flask_module.request.args.get('attempt_list_id')  # type: Optional[str]
+            res = self._gateway_controller.trigger_attemptlist_retry(attempt_list_id)
+            if res is True:
```

Comment:
If the return value is already a boolean, you may just write `if res`.
Please rename res to something like `retryWasSuccessful`.